.github/workflows/index-resources.yml

```diff
@@ -10,8 +10,45 @@ on:
   # Manually triggered using GitHub's UI
   workflow_dispatch:
 
+  workflow_call:
+
+
+defaults:
+  run:
+    # This is the same as GitHub Action's `bash` keyword as of 20 June 2023:
+    # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell
+    #
+    # Completely spelling it out here so that GitHub can't change it out from under us
+    # and we don't have to refer to the docs to know the expected behavior.
+    shell: bash --noprofile --norc -eo pipefail {0}
+
 jobs:
+  build-ref-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      ref-matrix: ${{ steps.ref-matrix.outputs.ref-matrix }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: 'package.json'
+      - id: ref-matrix
+        name: Create ref matrix
+        env:
+          HEROKU_TOKEN: ${{ secrets.HEROKU_TOKEN_READ_PROTECTED }}
+        run: |
+          echo "ref-matrix=$(./scripts/get-resource-index-ref-matrix)" | tee -a "$GITHUB_OUTPUT"
```

Comment:
This job unfortunately ends up [not failing](https://github.com/nextstrain/nextstrain.org/actions/runs/9116694184/job/25066044977#step:4:7) even when `./scripts/get-resource-index-ref-matrix` errors because the error exit code is ignored within the process substitution, e.g.

```console
$ if (set -euo pipefail; echo "x=$(exit 1)"); then echo ok; else echo error; fi
x=
ok
```

```bash
(echo -n "ref-matrix="; ./scripts/get-resource-index-ref-matrix) | tee -a "$GITHUB_OUTPUT"
```

would work as a substitute.

Reply:
Thanks for flagging that, didn't realize this would be an issue. 
I think might clearer to just assign to a variable then:

```shell
ref_matrix=$(./scripts/get-resource-index-ref-matrix)
echo "ref_matrix=$ref_matrix" | tee -a "$GITHUB_OUTPUT"
```

Reply:
Updating in https://github.com/nextstrain/nextstrain.org/pull/866